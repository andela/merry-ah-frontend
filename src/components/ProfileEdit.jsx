import React from 'react';
import PropTypes from 'prop-types';

/**
 * ProfileEdit Component
 */

export const ProfileEdit = ({ user, handleEditProfile, handleChange }) => (
        <div className="p-2 m-3 col-12">
          <div className="row container" id="art-list">
            <p className="profile-header col-6 offset-3 text-center mt-5">
              Update Profile
            </p>
            <form className="col-md-8 offset-2" onSubmit={handleEditProfile}>
              <div className="form-group row">
                <label htmlFor="bio" className="col-sm-2 col-form-label">
                  Biography
                </label>
                <div className="col-sm-10">
                  <input
                    type="text"
                    className="form-control"
                    id="bio"
                    placeholder="Bio"
                    onChange={handleChange}
                  />
                </div>
              </div>
              <fieldset className="form-group">
                <div className="row">
                  <legend className="col-form-label col-sm-2 pt-1" >
                    User Type
                  </legend>
                  <div className="col-sm-10">
                    <div className="form-check">
                      <input
                        className="form-check-input"
                        type="radio"
                        name="gridRadios"
                        id="userType"
                        value="artist"
                        onChange={handleChange}
                      />
                      <label
                        className="form-check-label"
                        htmlFor="userType"
                      >
                        Artist
                      </label>
                    </div>
                    <div className="form-check disabled">
                      <input
                        className="form-check-input"
                        type="radio"
                        name="gridRadios"
                        id="userType"
                        value="user"
                        onChange={handleChange}
                      />
                      <label
                        className="form-check-label"
                        htmlFor="userType"
                      >
                        User
                      </label>
                    </div>
                  </div>
                </div>
              </fieldset>
              <div className="form-group row">
                <label htmlFor="image" className="col-sm-2 col-form-label">
                  Image
                </label>
                <div className="col-sm-10">
                  <input
                    type="file"
                    className="form-control"
                    id="image"
                    placeholder="Upload profile image"
                    onChange={handleChange}
                  />
                </div>
                </div>
                <div className="form-group row">
                <button
                disabled={user.isLoading}
                className="btn btn-primary col-sm-10 offset-2"
                type="submit">
                {!user.isLoading ? 'Proceed' : 'Updating' }
                </button>
                </div>
            </form>
          </div>
        </div>
);

ProfileEdit.propTypes = {
  user: PropTypes.shape({
    isLoading: PropTypes.bool,
  }),
  handleEditProfile: PropTypes.func,
  handleChange: PropTypes.func,
};
export default ProfileEdit;
