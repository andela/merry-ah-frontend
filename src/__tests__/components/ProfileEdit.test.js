/* eslint-disable import/no-unresolved */
import 'babel-polyfill';
import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ProfileEdit from '../../components/ProfileEdit.jsx';
import { props } from '../__mocks__/usersMock';

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <MemoryRouter>
      <ProfileEdit {...props} />
    </MemoryRouter>,
  );
});
describe('ProfileEdit Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should display laoding when the update is running', () => {
    const newProps = {
      user: {
        ...props.user,
        isLoading: true,
      },
    };
    wrapper = mount(
        <MemoryRouter>
          <ProfileEdit {...newProps} />
        </MemoryRouter>,
    );
    expect(wrapper.find('button').at(0).text()).toEqual('Updating');
  });
});
