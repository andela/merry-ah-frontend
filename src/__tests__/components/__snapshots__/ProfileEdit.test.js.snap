// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileEdit Component should match snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "mxc061",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "mxc061",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ProfileEdit
      auth={
        Object {
          "user": Object {
            "id": 7,
          },
        }
      }
      editProfile={[Function]}
      getUserArts={[Function]}
      getUserProfile={[Function]}
      user={
        Object {
          "isGettingArts": false,
          "userArts": Array [
            Object {
              "Category": Object {
                "categoryName": "Poetry",
              },
              "description": "some descriptions",
              "featuredImg": "image.url",
              "slug": "isuguixu",
              "title": "some title",
            },
          ],
          "userProfile": Object {
            "id": 7,
            "profile": Object {
              "bio": "my bio",
              "firstName": "Juliet",
              "imgURL": "img.URL",
              "lastName": "Ezekwe",
            },
            "userType": "artist",
          },
        }
      }
    >
      <div
        className="p-2 m-3 col-12"
      >
        <div
          className="row container"
          id="art-list"
        >
          <p
            className="profile-header col-6 offset-3 text-center mt-5"
          >
            Update Profile
          </p>
          <form
            className="col-md-8 offset-2"
          >
            <div
              className="form-group row"
            >
              <label
                className="col-sm-2 col-form-label"
                htmlFor="bio"
              >
                Biography
              </label>
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control"
                  id="bio"
                  placeholder="Bio"
                  type="text"
                />
              </div>
            </div>
            <fieldset
              className="form-group"
            >
              <div
                className="row"
              >
                <legend
                  className="col-form-label col-sm-2 pt-1"
                >
                  User Type
                </legend>
                <div
                  className="col-sm-10"
                >
                  <div
                    className="form-check"
                  >
                    <input
                      className="form-check-input"
                      id="userType"
                      name="gridRadios"
                      type="radio"
                      value="artist"
                    />
                    <label
                      className="form-check-label"
                      htmlFor="userType"
                    >
                      Artist
                    </label>
                  </div>
                  <div
                    className="form-check disabled"
                  >
                    <input
                      className="form-check-input"
                      id="userType"
                      name="gridRadios"
                      type="radio"
                      value="user"
                    />
                    <label
                      className="form-check-label"
                      htmlFor="userType"
                    >
                      User
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
            <div
              className="form-group row"
            >
              <label
                className="col-sm-2 col-form-label"
                htmlFor="image"
              >
                Image
              </label>
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control"
                  id="image"
                  placeholder="Upload profile image"
                  type="file"
                />
              </div>
            </div>
            <div
              className="form-group row"
            >
              <button
                className="btn btn-primary col-sm-10 offset-2"
                type="submit"
              >
                Proceed
              </button>
            </div>
          </form>
        </div>
      </div>
    </ProfileEdit>
  </Router>
</MemoryRouter>
`;
